src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   * 
+   * @return std::vector<real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<real*> GetRedecompResponsePtrs()
+  {
+    return ParentField::GetRedecompFieldPtrs(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the redecomp mesh
+   * 
+   * @return const mfem::GridFunction& 
+   */
+  const mfem::GridFunction& GetRedecompResponse() const
+  {
+    return redecomp_response_;
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the parent mesh
+   * 
+   * @return mfem::ParGridFunction 
+   */
+  mfem::ParGridFunction GetParentResponse() const
+  {
+    return GetParentTransfer().RedecompToParent(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the parent to redecomp transfer object
+   * 
+   * @return const ParentRedecompTransfer& 
+   */
+  const ParentRedecompTransfer& GetParentTransfer() const
+  { 
+    return GetUpdateData().vector_xfer_;
+  }
+
+  /**
+   * @brief Add/replace the parent velocity grid function
+   * 
+   * @param velocity Velocity grid function on the parent mesh
+   */
+  void SetParentVelocity(const mfem::ParGridFunction& velocity);
+
+  /**
+   * @brief Determine if a velocity grid function has been set
+   * 
+   * @return true: Velocity grid function has been set
+   * @return false: Velocity grid function has not been set
+   */
+  bool HasVelocity() const { return velocity_ != nullptr; }
+
+  /**
+   * @brief Get pointers to component arrays of the velocity on the redecomp mesh
+   * 
+   * @return std::vector<const real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<const real*> GetRedecompVelocityPtrs() const
+  {
+    return velocity_->GetRedecompFieldPtrs();
+  }
+
+  /**
+   * @brief Get the map from mesh 1 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap1() const
+  {
+    return GetUpdateData().elem_map_1_;
+  }
+
+  /**
+   * @brief Get the map from mesh 2 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap2() const
+  {
+    return GetUpdateData().elem_map_2_;
+  }
+
+  /**
+   * @brief Get the (coarse) boundary submesh containing contact surfaces
+   * 
+   * @return mfem::ParSubMesh& 
+   */
+  mfem::ParSubMesh& GetSubmesh() { return submesh_; }
+
+  /**
+   * @brief Get the (coarse) boundary submesh containing contact surfaces
+   * 
+   * @return const mfem::ParSubMesh& 
+   */
+  const mfem::ParSubMesh& GetSubmesh() const { return submesh_; }
+
+  /**
+   * @brief Get the (refined) boundary submesh containing contact surfaces
+   * 
+   * @return mfem::ParMesh*
+   */
+  mfem::ParMesh* GetLORSubmesh() { return lor_submesh_.get(); }
+
+  /**
+   * @brief Get the refined boundary submesh containing contact surfaces
```

Comment:
Replace 'refined' with LOR or low order refined to be very clear as to how/why it is 'refined'. Same question as above, is there a single LOR Submesh containing both contact surfaces in a coupling scheme interaction?

Reply:
I think refined is clearer; there's no notion of "order" on a mesh, just a grid function.

Reply:
Will update this to LOR based on our discussion.